src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -3465,38 +4060,25 @@ class CgroupUtils(object):
         """
         pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Method called" % caller_name())
         pbs.logmsg(pbs.EVENT_DEBUG4, "Local jobs: %s" % (local_jobs))
-        new_jobs = set()
-        mom_priv_jobs_dir = os.path.join(PBS_MOM_HOME, 'mom_priv', 'jobs')
-        # Make a list of jobs from jobids in cgroup_jobs file and .JB files.
-        # These jobs are new since local_jobs was written to mom hook input
-        # file. We should not mistake their cgroups as orphans
-        cgroup_jobs = self.read_cgroup_jobs()
-        pbs.logmsg(pbs.EVENT_DEBUG4,
-                   'cgroup_jobs file content: %s' % cgroup_jobs)
-        if cgroup_jobs:
-            for jobid in cgroup_jobs:
-                new_jobs.add(jobid)
-            self.empty_cgroup_jobs_file()
-        for p in glob.glob(os.path.join(mom_priv_jobs_dir, '*.JB')):
-            pbs.logmsg(pbs.EVENT_DEBUG4, 'Found file %s' % p)
-            jobid = os.path.splitext(os.path.basename(p))[0]
-            new_jobs.add(jobid)
-        pbs.logmsg(pbs.EVENT_DEBUG4, 'New jobs: %s' % str(new_jobs))
         remaining = 0
         for key in self.paths:
             path = os.path.dirname(self._cgroup_path(key))
-            for subdir in glob.glob(os.path.join(path, '[0-9]*')):
-                jobid = os.path.basename(subdir)
-                if jobid not in local_jobs and jobid not in new_jobs:
-                    if not jobid.endswith('-orphan'):
-                        try:
-                            os.rename(subdir, subdir + '-orphan')
-                        except KeyboardInterrupt:
-                            raise
-                        except:
-                            pass
-            for subdir in glob.glob(os.path.join(path, '[0-9]*-orphan')):
-                jobid = os.path.basename(subdir)
+            # Identify any orphans and append an orphan suffix
+            pattern = self._systemd_subdir_wildcard()
+            for subdir in glob.glob(os.path.join(path, pattern)):
+                jobid = self._systemd_subdir_to_jobid(os.path.basename(subdir))
+                if jobid in local_jobs or jobid.endswith('.orphan'):
+                    continue
+                try:
+                    os.rename(subdir, subdir + '.orphan')
+                except KeyboardInterrupt:
+                    raise
+                except:
+                    pass
```

Comment:
Don't use blanket except statements like this.  Narrow the scope.

Reply:
This is another case where we don't care what the error was, or even that it occurred. We just keep trying to identify and cleanup orphans. I added a log message in the case where os.rename() fails.

Reply:
Using blanket except pass statements is a bad habit to get into.  I understand you don't care about the exceptions, but I'd still suggest you narrow the scope.  You know what os.rename() will throw.